src/features/hotspots/root/WelcomeOverview.tsx

```diff
@@ -78,6 +82,56 @@ const WelcomeOverview = ({ accountRewards }: Props) => {
     (state: RootState) => state.validators.validators.loading,
   )
 
+  const balance = useSelector(
+    (state: RootState) => state.account.account?.balance?.integerBalance,
+  )
+
+  const hotspotRewards = useSelector(
+    (state: RootState) => state.hotspots.rewards || {},
+  )
+
+  const total = useSelector(
+    (state: RootState) => state.account.rewardsSum.total,
+  )
+
+  const currentOraclePrice = useSelector(
+    (state: RootState) => state.heliumData.currentOraclePrice,
+  )
+
+  // Hook that is used for helium balance widget.
+  useAsync(async () => {
+    const oraclePrice = currentOraclePrice?.price
+    const floatBalance = oraclePrice?.floatBalance
+
+    await SharedGroupPreferences.setItem(
+      'myBalanceWidgetKey',
+      {
+        hntPrice: floatBalance,
+        hntDailyEarnings: total,
+        balance,
+      },
+      widgetGroup,
+    )
+  }, [balance, currentOraclePrice, total])
+
+  // Hook that is used for helium hotspots widget.
+  useAsync(async () => {
```

Comment:
This won't update unless the app is in the foreground. How often does the widget refresh the data?

Reply:
The widget refreshes every 5 mins